packages/types/src/block-kit/blocks.ts

```diff
@@ -1,68 +1,202 @@
+   * Maximum length for the text in this field is 2000 characters.
+   */
   label: PlainTextElement;
+  /**
+   * @description An optional hint that appears below an input element in a lighter grey. It must be a
+   * {@link PlainTextElement object}. Maximum length for the `text` in this field is 2000 characters.
+   */
   hint?: PlainTextElement;
+  /**
+   * @description A boolean that indicates whether the input element may be empty when a user submits the modal.
+   * Defaults to `false`.
+   */
   optional?: boolean;
+  /**
+   * @description A block element.
+   */
   element: Select | MultiSelect | Datepicker | Timepicker | DateTimepicker | PlainTextInput | URLInput | EmailInput
   | NumberInput | RadioButtons | Checkboxes;
+  /**
+   * @description A boolean that indicates whether or not the use of elements in this block should dispatch a
+   * {@link https://api.slack.com/reference/interaction-payloads/block-actions block_actions payload}. Defaults to `false`.
+   */
   dispatch_action?: boolean;
 }
 
+// TODO: breaking change: use a discriminative union to represent section block using `text` or `fields` but
+// not both or neither.
+/**
+ * @description Displays text, possibly alongside block elements. A section can be used as a simple text block, in
+ * combination with text fields, or side-by-side with certain
+ * {@link https://api.slack.com/reference/messaging/block-elements block elements}.
+ * @see {@link https://api.slack.com/reference/block-kit/blocks#section Section block reference}.
+ */
 export interface SectionBlock extends Block {
+  /**
+   * @description The type of block. For a section block, `type` is always `section`.
+   */
   type: 'section';
-  text?: PlainTextElement | MrkdwnElement; // either this or fields must be defined
+  /**
+   * @description The text for the block, in the form of a text object. Minimum length for the `text` in this field is
+   * 1 and maximum length is 3000 characters. This field is not required if a valid array of `fields` objects is
+   * provided instead.
+   */
+  text?: PlainTextElement | MrkdwnElement;
+  /**
+   * @description Required if no `text` is provided. An array of text objects. Any text objects included with `fields`
+   * will be rendered in a compact format that allows for 2 columns of side-by-side text. Maximum number of items is 10.
+   * Maximum length for the text in each item is 2000 characters.
+   * {@link https://app.slack.com/block-kit-builder/T029V6468RL#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22text%22:%22A%20message%20*with%20some%20bold%20text*%20and%20_some%20italicized%20text_.%22,%22type%22:%22mrkdwn%22%7D,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Priority*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Type*%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22High%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22String%22%7D%5D%7D%5D%7D Click here for an example}.
```

Comment:
You can remove a team_id from the URL
```suggestion
   * {@link https://app.slack.com/block-kit-builder/#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22text%22:%22A%20message%20*with%20some%20bold%20text*%20and%20_some%20italicized%20text_.%22,%22type%22:%22mrkdwn%22%7D,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Priority*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Type*%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22High%22%7D,%7B%22type%22:%22plain_text%22,%22text%22:%22String%22%7D%5D%7D%5D%7D Click here for an example}.
```